paasta_tools/kubernetes/remote_run.py

```diff
@@ -244,6 +284,45 @@ def remote_run_token(
     return create_temp_exec_token(kube_client, namespace, service_account)
 
 
+def generate_toolbox_deployment(
+    service: str, cluster: str, user: str
+) -> EksDeploymentConfig:
+    """Creates virtual EKS deployment for toolbox containers starting from adhoc configuration
+
+    :param str service: toolbox name
+    :param str cluster: target deployment cluster
+    :param str user: user requesting the toolbox
+    :return: deployment configuration
+    """
+    if not user.isalnum():
+        raise RemoteRunError(
+            f"Provided username contains non-alphanumeric characters: {user}"
+        )
+    adhoc_instance = service[len(TOOLBOX_MOCK_SERVICE) + 1 :]
+    adhoc_deployment = load_adhoc_job_config(
+        TOOLBOX_MOCK_SERVICE,
+        adhoc_instance,
+        cluster,
+        load_deployments=False,
+    )
+    adhoc_deployment.config_dict.setdefault("extra_volumes", []).append(
+        {
+            "containerPath": f"/etc/authorized_keys.d/{user}.pub",
+            "hostPath": f"/etc/authorized_keys.d/{user}.pub",
+            "mode": "RO",
+        },
+    )
+    adhoc_deployment.config_dict.setdefault("env", {})["SANDBOX_USER"] = user
+    adhoc_deployment.config_dict["routable_ip"] = True
+    return EksDeploymentConfig(
+        service=service,
+        cluster=cluster,
+        instance="main",
```

Comment:
thoughts on having the `instance` here be the toolkit instance + the username? if i'm not mistaken, that would make it easier to filter/view who is using what with `kubectl get pods -L paasta.yelp.com/instance`

Reply:
(unless that has other side-effects i'm not thinking of)

Reply:
There would be a bit much redundancy in the pod name. It'd basically container the user name twice. I think we can filter things decently enough already with the `job-name` label.